src/storage/src/source/mysql/replication.rs

```diff
@@ -462,15 +470,168 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
+                                    {
+                                        trace!(%id, "timely-{worker_id} DDL change \
+                                                        dropped table: {dropped_table:?}: {err:?}");
+                                        let (output_index, _) = &table_info[&dropped_table];
+                                        let gtid_cap = data_cap_set.delayed(new_gtid);
+                                        data_output
+                                            .give(
+                                                &gtid_cap,
+                                                ((*output_index, Err(err)), new_gtid.clone(), 1),
+                                            )
+                                            .await;
+                                        errored_tables.insert(dropped_table.clone());
+                                    }
+                                }
+
+                            // Detect `TRUNCATE [TABLE] <tbl>` statements
+                            } else if first.eq_ignore_ascii_case("truncate") {
+                                let table = if second.eq_ignore_ascii_case("table") {
+                                    table_ident(query_iter.next().ok_or_else(|| {
+                                        TransientError::Generic(anyhow::anyhow!(
+                                            "Invalid DDL query: {}",
+                                            query
+                                        ))
+                                    })?)?
+                                } else {
+                                    table_ident(second)?
+                                };
+                                if table_info.contains_key(&table) {
+                                    trace!(%id, "timely-{worker_id} TRUNCATE detected \
+                                                for {table:?}");
+                                    let (output_index, _) = &table_info[&table];
+                                    let gtid_cap = data_cap_set.delayed(new_gtid);
+                                    data_output
+                                        .give(
+                                            &gtid_cap,
+                                            (
+                                                (
+                                                    *output_index,
+                                                    Err(DefiniteError::TableTruncated(
+                                                        table.to_ast_string(),
+                                                    )),
+                                                ),
+                                                new_gtid.clone(),
+                                                1,
+                                            ),
+                                        )
+                                        .await;
```

Comment:
Can we tidy this call up a bit? Maybe pulling the `update` triplet out. But also, we've consumed a lot of our indentation budget at this point so I would consider Rainer's suggestion of splitting some things out